docs/components/HTTP.md

```diff
@@ -0,0 +1,468 @@
+* [HTTP](#http)
+  * [HTTP lifecycle endpoints](#http-lifecycle-endpoints)
+  * [HTTP Middlewares](#http-middlewares)
+    * [Middleware Chain](#middleware-chain)
+    * [Helper Middlewares](#helper-middlewares)
+  * [HTTP Routes](#http-routes)
+    * [HTTP Method](#http-method)
+    * [Processor](#processor)
+    * [Raw RouteBuilder Constructor](#raw-routebuilder-constructor)
+    * [Middlewares per Route](#middlewares-per-route)
+    * [Security](#security)
+    * [Tracing](#tracing)
+    * [HTTP Caching](#http-caching)
+
+# HTTP
+
+The HTTP component provides the functionality for creating an HTTP server exposing the relevant routes. 
+It wraps the logic and handles the boilerplate for the `net.http` go package.
+
+The way to initialise an HTTP component is through the `patron http.Builder` struct.
+```go
+// NewBuilder initiates the HTTP component builder chain.
+// The builder instantiates the component using default values for
+// HTTP Port, Alive/Ready check functions and Read/Write timeouts.
+func NewBuilder() *Builder {
```

Comment:
@drakos74 
Maintaining that much of details will be an issue.
For the low level techincal doc we can use the build in https://pkg.go.dev/github.com/beatlabs/patron.
What do you think?

Reply:
Yes, good point. Should i replace this code segment with a shoprt example of using the builder ?

Reply:
yes, that would make sense.